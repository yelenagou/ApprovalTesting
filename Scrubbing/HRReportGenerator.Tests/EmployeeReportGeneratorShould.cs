using System;
using Xunit;
using ApprovalTests;
using ApprovalTests.Reporters;
using System.Text.RegularExpressions;

namespace HRReportGenerator.Tests
{
    public class EmployeeReportGeneratorShould
    {
        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void GenerateCorrectReportWhenMultipleEmployees()
        {
            var sut = new EmployeeReportGenerator();

            var employees = new[]
            {
                new Employee {FirstName = "Sarah", LastName = "Smith", PayGrade=1, DateOfBirth = new DateTime(1980,4,10)},
                new Employee {FirstName = "Gentry", LastName = "Smith", PayGrade=3, DateOfBirth = new DateTime(1970,7,1)},
                new Employee {FirstName = "Arnold", LastName = "Jones", PayGrade=2, DateOfBirth = new DateTime(1972,12,21)}
            };

            string reportText = sut.Generate(employees);

            Approvals.Verify(reportText,
                (input) => Regex.Replace(input, "Date Generated.*", "Date Generated: July 1, 2000"));
        }
    }
}
