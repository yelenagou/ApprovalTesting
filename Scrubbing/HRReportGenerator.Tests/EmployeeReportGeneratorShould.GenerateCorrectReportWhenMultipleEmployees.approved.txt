HR Employees Report
-------------------

Date Generated: July 1, 2000

First Name               Last Name                  Pay Grade     DOB
Sarah                    Smith                      1             April 10, 1980
Gentry                   Smith                      3             July 1, 1970
Arnold                   Jones                      2             December 21, 1972

Total Employees: 3